spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/DistriParameterSynchronizer.scala

```diff
@@ -153,26 +127,25 @@ class BlockManagerParameterSynchronizer[T: ClassTag](partitionID: Int,
       }
     })
 
-  // thread pool to update sow on fetching completion
-  private val fetchCompletionPool: ExecutorService = Executors.
-    newFixedThreadPool(fetchCompletionPoolSize,
-    new ThreadFactory {
-      override def newThread(r: Runnable): Thread = {
-        val t = Executors.defaultThreadFactory().newThread(r)
-        t.setDaemon(true)
-        t
+  (0 until syncPoolSize).map(wp => {
+    fetchPool.submit(new Runnable {
+      override def run(): Unit = {
+        val v = threadCount.incrementAndGet()
+        Affinity.setAffinity(communicationStartCore + (v) % 4)
       }
     })
+  })
```

Comment:
I think this block of code should be into a method.

Reply:
ok